wire/internal/wire/analyze.go

```diff
@@ -226,14 +230,71 @@ dfs:
 	if len(ec.errors) > 0 {
 		return nil, ec.errors
 	}
+	verifyArgsUsed(set, used, ec)
+	if len(ec.errors) > 0 {
+		return nil, ec.errors
+	}
 	return calls, nil
 }
 
-// buildProviderMap creates the providerMap field for a given provider set.
-// The given provider set's providerMap field is ignored.
-func buildProviderMap(fset *token.FileSet, hasher typeutil.Hasher, set *ProviderSet) (*typeutil.Map, []error) {
+func verifyArgsUsed(set *ProviderSet, used []*providerSetSrc, ec *errorCollector) {
```

Comment:
Instead of taking in an `*errorCollector`, return `[]error`.

Reply:
Done.